pkg/secrethub/credentials/key.go

```diff
@@ -56,26 +56,37 @@ func (k Key) Export() ([]byte, error) {
 	return EncodeCredential(k.key)
 }
 
-// ImportKey returns a Key by loading it from the provided credentialReader.
-// If the key is encrypted with a passphrase, passphraseReader should be provided. This is used to read a passphrase
-// from that is used for decryption. If the passphrase is incorrect, a new passphrase will be read up to 3 times.
-func ImportKey(credentialReader, passphraseReader Reader) (Key, error) {
-	bytes, err := credentialReader.Read()
-	if err != nil {
-		return Key{}, err
+type KeyDecoder interface {
```

Comment:
Please suggest any other more suitable names for this interface. It is currently quite easy to confuse it with the `Decoder` interface which decodes RSA keys. This one decodes the entire credential so maybe something like `CredentialDecoder` would work better, but I'm not sure because it returns the `Key` type.

Reply:
See my other comment on not exposing the decoder to the user, as always the same decoder is used.

If we do that, we can remove this interface entirely, or make it private.